File: ABC/282/b.cpp
#include <iostream>
#include <vector>
using namespace std;

using ll = long long;

int main() {
    int n, m;
    cin >> n >> m;

    vector<string> s(n);
    for (int i = 0; i < n; ++i) cin >> s[i];

    int ans = 0;
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < n; ++j) {
            if (i == j) continue;
            bool ok = true;
            for (int k = 0; k < m; ++k) {
                if (s[i][k] == 'x' && s[j][k] == 'x') ok = false;
            }
            if (ok) ans++;
        }
    }
    cout << ans / 2 << endl;

    return 0;
}

File: ABC/282/b.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let input = [];
rl.on('line', (line) => {
    input.push(line);
}).on('close', () => {
    const [n, m] = input[0].split(' ').map(Number);
    let s = input.slice(1, n + 1);

    let ans = 0;

    for (let i = 0; i < n; ++i) {
        for (let j = i + 1; j < n; ++j) {
            let ok = true;
            for (let k = 0; k < m; ++k) {
                if (s[i][k] == 'x' && s[j][k] == 'x') ok = false;
            }
            if (ok) ans++;
        }
    }
    console.log(ans);

    process.exit(0);
});
